addons/mail/static/src/models/message_reaction/message_reaction.js

```diff
@@ -0,0 +1,72 @@
+/** @odoo-module **/
+
+import { registerNewModel } from '@mail/model/model_core';
+import { attr, many2one, one2many} from '@mail/model/model_field';
+import { clear, insert, insertAndReplace, link, replace, unlink, unlinkAll } from '@mail/model/model_field_command';
+
+function factory(dependencies) {
+
+    class MessageReaction extends dependencies['mail.model'] {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {Object} data
+         * @return {Object}
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('emoji_unicode' in data) {
+                data2.emoji_unicode = data.emoji_unicode;
+            }
+            if ('id' in data) {
+                data2.id = data.id;
+            }
+            if ('message_id' in data) {
+                data2.message = link({
+                    id: data.message_id,
+                });
+            }
+            if ('reactor' in data) {
+                data2.reactor = data.reactor;
+            }
+            return data2;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static _createRecordLocalId(data) {
+            return `${this.modelName}_${data.id}`;
+        }
+    }
+
+    MessageReaction.fields = {
+        emoji_unicode: attr({
+            required: true,
+        }),
+        id: attr({
+            required: true,
+        }),
+        message: many2one('mail.message', {
+            inverse: 'reactions',
+            required: true,
+        }),
+        reactor: many2one('mail.partner', {
```

Comment:
Humans are radioactive, but I don't think it's appropriate to call partner "reactors" (FYI this designation is specific to nuclear physics).

I think the proper name would be "responder".

Reply:
Totally disagree. Despite the name "reactor" can make think of a joke, the first definition of a reactor is "a one that reacts" which is exactly the case here. 
Also, "responder" makes me think to someone who responds (with a textual response, not a reaction). 

I think we can agree to create a neologism "reactioner" to have both meaning of the one who reacts and the one who responds.